plugins/formal-verification/formver.annotations/src/org/jetbrains/kotlin/formver/plugin/Builtins.kt

```diff
@@ -10,4 +10,10 @@ package org.jetbrains.kotlin.formver.plugin
  * This function hooks-in in the `formver` plugin, its invocation in a Kotlin
  * program does not do anything.
  */
-fun verify(@Suppress("UNUSED_PARAMETER") vararg predicates: Boolean) = Unit
\ No newline at end of file
+fun verify(@Suppress("UNUSED_PARAMETER") vararg predicates: Boolean) = Unit
+
+fun loopInvariants(@Suppress("UNUSED_PARAMETER") body: InvariantBuilder.() -> Unit) = Unit
+
+class InvariantBuilder {
+    operator fun Boolean.unaryPlus() = Unit
```

Comment:
So I'm not sure having a unary plus here makes sense.  The benefit is that we know the result is a Boolean, and that we're less likely to get unused expression warnings.  WDYT?

Reply:
I agree (will get rid of this `unaryPlus`). 
We can report our own errors if the result is not boolean.

Reply:
Not sure how to avoid unused expressions without changing the `kotlinc` code though.